### PR TITLE
Fix tf/tfstate output for security groups

### PR DESCRIPTION
**Pull request by woohgit**
Adjusted the tests and both the tf/tfstate exports according to the statements
below. The tests should be improved too, for now they're ok.

Note that if it's a VPC security group, the group name is nil in the permission
set.

 EC2-Classic security groups can be:

  - security groups from the same account
  - security groups from another account
  - CIDR

If it is an EC2-Classic security group from the same account, the rule should
contain only the security group name not the group id.

Example tf export snippet: `security_groups = ["sg-1234efgh"]`

If it's an EC2-Classic security group from another account, the role should
contain the account id and the security group name, not the id.

Example tf export snippet: `security_groups = ["098765432109/sg-1234efgh"]`

VPC security groups can be:
  - security groups from the same account + same VPC
  - CIDR

A rule inside of a VPC security group can't be SGs from a different VPC or from
a differert account. Only CIDRs and SGs from the same VPC.

A rule should contain only the security group ID or a CIDR.

Example tf export snippet: `security_groups = ["sg-1234efgh"]`

The [documentation](https://www.terraform.io/docs/providers/aws/r/security_group.html) says:

```
security_groups - (Optional)
List of security group Group Names if using EC2-Classic or the default VPC, or
Group IDs if using a non-default VPC.
```

### CODE REVIEW

**Comment by woohgit** [comment]
Note that I've tested this in an environment with:
- EC2-Classic only
- VPC only
- Mixed

The outputs were always OK and the tf was in sync with the tfstate.

File: lib/terraforming/resource/security_group.rb [hunk 1]
```diff
@@ -138,21 +139,32 @@ def permission_hashcode_of(security_group, permission)
...
         @client.describe_security_groups.security_groups
       end
 
-      def security_groups_in(permission)
-        permission.user_id_group_pairs.map { |range| range.group_id }
+      def security_groups_in(permission, security_group)
+        permission.user_id_group_pairs.map { |range|
+          # EC2-Classic, same account
+          if security_group.owner_id == range.user_id and !range.group_name.nil?
+            next range.group_name
```
Comment (dtan4) [inline lib/terraforming/resource/security_group.rb]: `next` is unnecessary. Please remove.

File: lib/terraforming/resource/security_group.rb [hunk 2]
```diff
@@ -138,21 +139,32 @@ def permission_hashcode_of(security_group, permission)
...
       def security_groups
         @client.describe_security_groups.security_groups
       end
 
-      def security_groups_in(permission)
-        permission.user_id_group_pairs.map { |range| range.group_id }
+      def security_groups_in(permission, security_group)
+        permission.user_id_group_pairs.map { |range|
+          # EC2-Classic, same account
+          if security_group.owner_id == range.user_id and !range.group_name.nil?
```
Comment (dtan4) [inline lib/terraforming/resource/security_group.rb]: Replace `and` to `&&`.

**Comment by woohgit** [comment]
Replaced all `and` to `&&` and removed all `next` occurrences.

**Comment by dtan4** [comment]
(I confirmed that CI fails are not related to your changes.)

LGTM 👍 Thanks ❗ 
Sorry for being late.

**Pull request merged**